### PR TITLE
service/share: Remove unused `Service` interface and replace it with struct instead

### PR DESCRIPTION
**Pull request by renaynay**
This PR removes the unused `Service` interface and replaces it with the struct instead.

### CODE REVIEW

**Comment by Wondertan** [comment]
@renaynay, pls base it on top of #693, so that this PR does not show changes from it

File: node/rpc/component.go [hunk 1]
```diff
@@ -2,18 +2,34 @@ package rpc
...
 			OnStart: serv.Start,
 			OnStop:  serv.Stop,
 		})
 		return serv
 	}
 }
+
+func HandlerComponents(
+	state *state.Service,
+	share share.Service,
```
Comment (vgonkivs) [inline node/rpc/component.go:28]: should not it also be a pointer?

**Comment by Wondertan** [comment]
Because of the work, iirc
> Github prevents it

**Comment by renaynay** [comment]
Closing in favour of #707 due to hairy merge conflicts.

**Pull request closed without merging**